db/db_impl/db_impl_write.cc

```diff
@@ -1192,6 +1193,11 @@ IOStatus DBImpl::WriteToWAL(const WriteBatch& merged_batch,
   assert(log_size != nullptr);
 
   Slice log_entry = WriteBatchInternal::Contents(&merged_batch);
+  TEST_SYNC_POINT_CALLBACK("DBImpl::WriteToWAL:log_entry", &log_entry);
```

Comment:
Does it still detect in the same place if the corruption happens on pre-merged batches?

Reply:
Good catch: no, for example when a follower's write batch has it count field corrupted, [this](https://github.com/facebook/rocksdb/blob/f85b31a2e958a65e73797932f23985e398562aa7/db/write_batch.cc#L2794) copy will go over vector boundary and fails [this](https://github.com/facebook/rocksdb/blob/f85b31a2e958a65e73797932f23985e398562aa7/util/autovector.h#L245) assert. 
I'm making changes to address this.